celery/backends/dynamodb.py

```diff
@@ -246,8 +290,11 @@ def _prepare_put_request(self, key, value):
                     self._value_field.data_type: value
                 },
                 self._timestamp_field.name: {
-                    self._timestamp_field.data_type: str(time())
-                }
+                    self._timestamp_field.data_type: str(_now)
+                },
+                self._ttl_field.name: {
+                    self._ttl_field.data_type: str(int(_now) + int(self.expires))
```

Comment:
Perhaps we should handle cases where `expires` is `None`/`0` which disables expiration, as seen in the [documentation](http://docs.celeryproject.org/en/latest/userguide/configuration.html#result-expires)? 

What happens if you set `NULL` as the TTL field value, if this is possible? Would that perhaps disable expiration for this document?

Additionally, DynamoDB should be added in the documentation list of result backends supporting expiration.